### PR TITLE
Migration to Gradle Version Catalog

### PR DESCRIPTION
**Pull request by donbeave**

### CODE REVIEW

**Comment by graemerocher** [comment]
@melix does it make sense for `testImplementation` dependency versions to go into the catalog definitions?

**Comment by melix** [comment]
> @melix does it make sense for testImplementation dependency versions to go into the catalog definitions?

Yes it does. The catalog should be the single place of truth where to put dependency versions.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+micronaut-jdbc = { module = "io.micronaut.sql:micronaut-jdbc" }
+micronaut-management = { module = "io.micronaut:micronaut-management" }
+micronaut-rxjava2 = { module = "io.micronaut.rxjava2:micronaut-rxjava2" }
+
+cache-api = { module = "javax.cache:cache-api", version.ref = "jcache" }
+groovy-json = { module = "org.codehaus.groovy:groovy-json", version.ref = "groovy" }
+grpc-api = { module = "io.grpc:grpc-api" }
+logback-classic = { module = "ch.qos.logback:logback-classic", version.ref = "logback-classic" }
+metrics-core = { module = "io.dropwizard.metrics:metrics-core", version.ref = "metrics-core" }
+micrometer-core = { module = "io.micrometer:micrometer-core", version.ref = "micrometer" }
```
Comment (melix) [inline gradle/libs.versions.toml:39]: Should those `micrometer-xxx` dependencies be actually managed in the Micronaut Micrometer BOM?

Comment (donbeave) [reply gradle/libs.versions.toml:39]: Maybe even better to use BOM directly: https://search.maven.org/artifact/io.micrometer/micrometer-bom/1.8.5/pom

**Comment by donbeave** [comment]
> > @melix does it make sense for testImplementation dependency versions to go into the catalog definitions?
> 
> Yes it does. The catalog should be the single place of truth where to put dependency versions.

I actually don't see the benefit of using Gradle Version Catalog for not versioned dependencies, dependencies that use the version from BOM package, for example:

```toml
micronaut-cache-core = { module = "io.micronaut.cache:micronaut-cache-core" }
```

Do you think it's better to leave them inside `build.gradle` file without change:

```groovy
dependencies {
    compileOnly 'io.micronaut.cache:micronaut-cache-core'
}
```

**Comment by melix** [comment]
> I actually don't see the benefit of using Gradle Version Catalog for not versioned dependencies, dependencies that use the version from BOM package, for example:

The goal of the version catalog is to have a centralized place where to put dependencies and type-safe accessors. Even if you don't use a _version_, it's still extremely useful. It avoids typos, and should you need to change the GAV coordinates of a library, you have a _single place_ where to do that.
In a nutshell, you should replace all strings with a "more strongly modeled" dependency notation.

**Comment by graemerocher** [comment]
Could you resolve the conflicts?

**Comment by donbeave** [comment]
> Could you resolve the conflicts?

Of course. In progress.

**Comment by sdelamo** [comment]
@timyates could you finish this PR

**Comment by donbeave** [comment]
I have pulled changes from the master and resolved conflicts.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+[versions]
+managed-micrometer = "1.9.0"
+
+groovy = "3.0.10"
+jcache = "1.1.1"
+logback-classic = "1.2.11"
+metrics-core = "4.2.9"
```
Comment (graemerocher) [inline gradle/libs.versions.toml:7]: should be prefixed with `managed-`

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+new-relic-registry = "0.8.0"
```
Comment (graemerocher) [inline gradle/libs.versions.toml:8]: should be prefixed with `managed-`

**Review by graemerocher — APPROVED** [review]

**Pull request merged**